spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerMethodTests.java

```diff
@@ -0,0 +1,190 @@
+			fail("Exception expected on multiple @Input method parameters");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("@Input annotation can not be used in multiple method parameters");
+		}
+	}
+
+	@Test
+	public void testMethodWithMultipleOutputParameters() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithMultipleOutputParameters.class, "--server.port=0");
+			fail("Exception expected on multiple @Output method parameters");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("@Output annotation can not be used in multiple method parameters");
+		}
+	}
+
+	@EnableBinding({Processor.class, AnotherOutput.class})
+	@EnableAutoConfiguration
+	public static class TestMethodWithMultipleOutputParameters {
+
+		@StreamListener
+		public void receive(@Input(Processor.INPUT) SubscribableChannel input, @Output(Processor.OUTPUT) final MessageChannel output1,
+				@Output(AnotherOutput.OUTPUT) final MessageChannel output2) {
+			input.subscribe(new MessageHandler() {
+				@Override
+				public void handleMessage(Message<?> message) throws MessagingException {
+					output1.send(org.springframework.messaging.support.MessageBuilder.withPayload(message.getPayload().toString().toUpperCase()).build());
+				}
+			});
+		}
+	}
+
+	@EnableBinding({Sink.class})
+	@EnableAutoConfiguration
+	public static class TestMethodWithoutInput {
+
+		@StreamListener
+		public void receive(FooPojo fooPojo) {
+		}
+	}
+
+	public static class FooPojo {
+
+		private String bar;
+
+		public String getBar() {
+			return this.bar;
+		}
+
+		public void setBar(String bar) {
+			this.bar = bar;
+		}
+	}
+
+	@EnableBinding({Sink.class})
+	@EnableAutoConfiguration
+	public static class TestMethodWithInputAsMethodAndParameter {
+
+		@StreamListener(Processor.INPUT)
+		public void receive(@Input(Processor.INPUT) FooPojo fooPojo, @Input(AnotherSink.INPUT) FooPojo fooPojo1) {
+		}
+	}
+
+	@EnableBinding({Processor.class, AnotherOutput.class})
+	@EnableAutoConfiguration
+	public static class TestMethodWithOutputAsMethodAndParameter {
+
+		@StreamListener
+		@Output(AnotherOutput.OUTPUT)
+		public void receive(@Input(Processor.INPUT) SubscribableChannel input, @Output(Processor.OUTPUT) final MessageChannel output1) {
+			input.subscribe(new MessageHandler() {
+				@Override
+				public void handleMessage(Message<?> message) throws MessagingException {
+					output1.send(org.springframework.messaging.support.MessageBuilder.withPayload(message.getPayload().toString().toUpperCase()).build());
+				}
+			});
+		}
+	}
+
+	@EnableBinding({Sink.class, AnotherSink.class})
+	@EnableAutoConfiguration
+	public static class TestMethodWithMultipleInputParameters {
+
+		@StreamListener
+		public void receive(@Input(Processor.INPUT) FooPojo fooPojo, @Input(AnotherSink.INPUT) FooPojo fooPojo1) {
+		}
+	}
+
+	public interface AnotherSink {
```

Comment:
We should consolidate around a set of classes that are used consistently during the test.